In [1]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore') 

##### Dataset: Restaurant Orders

In [2]:
res_orders = pd.read_csv('restaurant_orders.tsv', sep='\t')

In [3]:
res_orders

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

#### Q) What are the number of observations in the dataset? What are the number of columns in the dataset?

In [4]:
res_orders.shape

(4622, 5)

In [5]:
res_orders.shape[0]

4622

In [6]:
res_orders.shape[1]

5


4622 observations with 5 columns


In [7]:
res_orders.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

##### Q) How is the dataset indexed?

In [8]:
res_orders.index

RangeIndex(start=0, stop=4622, step=1)

##### Q) Which was the most-ordered item? 

In [9]:
res_orders

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

In [10]:
res_orders_groups = res_orders.groupby(by='item_name')

In [11]:
i=0
for item, group in res_orders_groups:
    print(item)
    print(group)
    print('---------------------------------------------------------------------------------')
    i = i+1
    if i == 3:
        break

6 Pack Soft Drink
      order_id  quantity          item_name choice_description item_price
298        129         1  6 Pack Soft Drink           [Sprite]     $6.49 
341        148         1  6 Pack Soft Drink        [Diet Coke]     $6.49 
357        154         1  6 Pack Soft Drink             [Coke]     $6.49 
388        168         1  6 Pack Soft Drink        [Diet Coke]     $6.49 
417        182         1  6 Pack Soft Drink        [Diet Coke]     $6.49 
553        230         1  6 Pack Soft Drink        [Diet Coke]     $6.49 
639        264         1  6 Pack Soft Drink        [Diet Coke]     $6.49 
721        298         1  6 Pack Soft Drink           [Nestea]     $6.49 
743        306         1  6 Pack Soft Drink             [Coke]     $6.49 
879        363         1  6 Pack Soft Drink             [Coke]     $6.49 
1026       422         1  6 Pack Soft Drink           [Sprite]     $6.49 
1051       432         1  6 Pack Soft Drink             [Coke]     $6.49 
1124       465      

In [12]:
res_orders_groups['quantity'].sum()

item_name
6 Pack Soft Drink                         55
Barbacoa Bowl                             66
Barbacoa Burrito                          91
Barbacoa Crispy Tacos                     12
Barbacoa Salad Bowl                       10
Barbacoa Soft Tacos                       25
Bottled Water                            211
Bowl                                       4
Burrito                                    6
Canned Soda                              126
Canned Soft Drink                        351
Carnitas Bowl                             71
Carnitas Burrito                          60
Carnitas Crispy Tacos                      8
Carnitas Salad                             1
Carnitas Salad Bowl                        6
Carnitas Soft Tacos                       40
Chicken Bowl                             761
Chicken Burrito                          591
Chicken Crispy Tacos                      50
Chicken Salad                              9
Chicken Salad Bowl                       123


In [13]:
res_orders_groups['quantity'].sum().max()

761

761 orders of Chicken Bowl

In [14]:
res_orders_groups['quantity'].value_counts()

item_name              quantity
6 Pack Soft Drink      1           53
                       2            1
Barbacoa Bowl          1           66
Barbacoa Burrito       1           91
Barbacoa Crispy Tacos  1           10
                                   ..
Veggie Crispy Tacos    1            1
Veggie Salad           1            6
Veggie Salad Bowl      1           18
Veggie Soft Tacos      1            6
                       2            1
Name: quantity, Length: 103, dtype: int64

In [15]:
quantity = res_orders_groups['quantity'].sum()

In [16]:
quantity.sort_values(ascending=False)

item_name
Chicken Bowl                             761
Chicken Burrito                          591
Chips and Guacamole                      506
Steak Burrito                            386
Canned Soft Drink                        351
Chips                                    230
Steak Bowl                               221
Bottled Water                            211
Chips and Fresh Tomato Salsa             130
Canned Soda                              126
Chicken Salad Bowl                       123
Chicken Soft Tacos                       120
Side of Chips                            110
Veggie Burrito                            97
Barbacoa Burrito                          91
Veggie Bowl                               87
Carnitas Bowl                             71
Barbacoa Bowl                             66
Carnitas Burrito                          60
Steak Soft Tacos                          56
6 Pack Soft Drink                         55
Chips and Tomatillo Red Chili Salsa       50


In [17]:
quantity.sort_values(ascending=False).head(1)

item_name
Chicken Bowl    761
Name: quantity, dtype: int64

Chicken Bowl is the most ordered item

In [18]:
quantity.max()

761

##### Q) For the most-ordered item, how many items were ordered?

In [19]:
quantity.max()

761

761 number of chicken bowls were ordered

##### Q) What was the most ordered item in the choice_description column?

In [20]:
res_orders

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

In [21]:
choice = res_orders.groupby(by='choice_description')

In [22]:
i=0
for c, group in choice:
    print(c)
    print(group)
    print('------------------------------------------------------------------------')
    i = i+1
    if i == 3:
        break

[Adobo-Marinated and Grilled Chicken, Pinto Beans, [Sour Cream, Salsa, Cheese, Cilantro-Lime Rice, Guacamole]]
     order_id  quantity item_name  \
510       214         1   Burrito   

                                    choice_description item_price  
510  [Adobo-Marinated and Grilled Chicken, Pinto Be...     $7.40   
------------------------------------------------------------------------
[Adobo-Marinated and Grilled Chicken, [Sour Cream, Cheese, Cilantro-Lime Rice]]
     order_id  quantity item_name  \
675       279         1   Burrito   

                                    choice_description item_price  
675  [Adobo-Marinated and Grilled Chicken, [Sour Cr...     $7.40   
------------------------------------------------------------------------
[Adobo-Marinated and Grilled Chicken]
     order_id  quantity     item_name                     choice_description  \
521       217         1  Crispy Tacos  [Adobo-Marinated and Grilled Chicken]   

    item_price  
521     $7.40   
--------

In [23]:
choice = choice['quantity'].sum()

In [24]:
choice.sort_values(ascending=False)

choice_description
[Diet Coke]                                                                                                                159
[Coke]                                                                                                                     143
[Sprite]                                                                                                                    89
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese, Sour Cream, Lettuce]]                                                      49
[Fresh Tomato Salsa, [Rice, Black Beans, Cheese, Sour Cream]]                                                               42
                                                                                                                          ... 
[Roasted Chili Corn Salsa, [Fajita Vegetables, Rice, Pinto Beans, Guacamole, Lettuce]]                                       1
[Roasted Chili Corn Salsa, [Fajita Vegetables, Rice, Sour Cream, Lettuce]]                  

In [25]:
choice.sort_values(ascending=False).head(1)

choice_description
[Diet Coke]    159
Name: quantity, dtype: int64

* Diet Coke is the most ordered item in the choice_description

##### Q) How many items were orderd in total?

In [26]:
res_orders

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

In [27]:
res_orders['quantity'].sum()

4972

4972 items were ordered in total

##### Q) Turn the item price into a float

In [28]:
res_orders

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

In [29]:
res_orders.dtypes

order_id               int64
quantity               int64
item_name             object
choice_description    object
item_price            object
dtype: object

In [30]:
res_orders['item_price'] = res_orders['item_price'].str.replace('$', '') 
res_orders['item_price'] = res_orders['item_price'].astype(float)

In [31]:
res_orders['item_price']

0        2.39
1        3.39
2        3.39
3        2.39
4       16.98
        ...  
4617    11.75
4618    11.75
4619    11.25
4620     8.75
4621     8.75
Name: item_price, Length: 4622, dtype: float64

##### Q) How many products cost more than $10.00?

In [32]:
# Q1. what are the products which cost more than $10.00?
# Q2. count of such items

In [33]:
res_orders['item_name'][res_orders['item_price'] > 10.00]

4             Chicken Bowl
5             Chicken Bowl
7            Steak Burrito
13            Chicken Bowl
23         Chicken Burrito
               ...        
4610         Steak Burrito
4611        Veggie Burrito
4617         Steak Burrito
4618         Steak Burrito
4619    Chicken Salad Bowl
Name: item_name, Length: 1130, dtype: object

In [34]:
res_orders['item_name'][res_orders['item_price'] > 10.00].nunique()

31

In [35]:
res_orders['item_name'][res_orders['item_price'] > 10.00].value_counts()

Chicken Bowl                    304
Chicken Burrito                 210
Steak Burrito                   139
Steak Bowl                       95
Chicken Salad Bowl               54
Veggie Bowl                      47
Veggie Burrito                   37
Carnitas Bowl                    32
Chicken Soft Tacos               30
Barbacoa Bowl                    27
Barbacoa Burrito                 24
Carnitas Burrito                 20
Steak Salad Bowl                 19
Chicken Crispy Tacos             18
Steak Crispy Tacos               12
Steak Soft Tacos                 10
Veggie Salad Bowl                10
Barbacoa Soft Tacos               8
Barbacoa Salad Bowl               5
Barbacoa Crispy Tacos             5
Carnitas Salad Bowl               4
Carnitas Soft Tacos               3
Veggie Soft Tacos                 3
Carnitas Crispy Tacos             3
Chips and Fresh Tomato Salsa      2
Chips and Guacamole               2
Bottled Water                     2
Chicken Salad               

31 products cost more than $10.00

##### Q) What was the quantity of the most expensive item ordered?


In [36]:
most_expensive = res_orders[res_orders['item_price'] == res_orders['item_price'].max()]
most_expensive

order_id  quantity                     item_name choice_description  \
3598      1443        15  Chips and Fresh Tomato Salsa                NaN   

      item_price  
3598       44.25

##### Q) How many times was a Veggie Salad Bowl ordered?

In [37]:
veggie = res_orders[res_orders['item_name'] == 'Veggie Salad Bowl']
veggie

order_id  quantity          item_name  \
186         83         1  Veggie Salad Bowl   
295        128         1  Veggie Salad Bowl   
455        195         1  Veggie Salad Bowl   
496        207         1  Veggie Salad Bowl   
960        394         1  Veggie Salad Bowl   
1316       536         1  Veggie Salad Bowl   
1884       760         1  Veggie Salad Bowl   
2156       869         1  Veggie Salad Bowl   
2223       896         1  Veggie Salad Bowl   
2269       913         1  Veggie Salad Bowl   
2683      1066         1  Veggie Salad Bowl   
3223      1289         1  Veggie Salad Bowl   
3293      1321         1  Veggie Salad Bowl   
4109      1646         1  Veggie Salad Bowl   
4201      1677         1  Veggie Salad Bowl   
4261      1700         1  Veggie Salad Bowl   
4541      1805         1  Veggie Salad Bowl   
4573      1818         1  Veggie Salad Bowl   

                                     choice_description  item_price  
186   [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       11.25  
295   [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...       11.25  
455   [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       11.25  
496   [Fresh Tomato Salsa, [Rice, Lettuce, Guacamole...       11.25  
960   [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...        8.75  
1316  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...        8.75  
1884  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       11.25  
2156  [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.25  
2223      [Roasted Chili Corn Salsa, Fajita Vegetables]        8.75  
2269  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...        8.75  
2683  [Roasted Chili Corn Salsa, [Fajita Vegetables,...        8.75  
3223  [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.25  
3293  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...        8.75  
4109  [Tomatillo Red Chili Salsa, [Fajita Vegetables...       11.25  
4201  [Fresh Tomato Salsa, [Fajita Vegetables, Black...       11.25  
4261  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       11.25  
4541  [Tomatillo Green Chili Salsa, [Fajita Vegetabl...        8.75  
4573  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...        8.75

In [38]:
len(veggie)

18

In [39]:
veggie['order_id'].count()

18

* Veggie Salad Bowl was ordered 18 times

##### Q) How many times did someone order more than one Canned Soda?

In [40]:
soda = res_orders[res_orders['item_name'] == 'Canned Soda']

In [41]:
more_than_one_soda = soda[soda['quantity'] > 1]

In [42]:
len(more_than_one_soda)

20

* 20 times more than one Canned Soda was ordered

##### Q) How much was the revenue for the period in the dataset?

In [43]:
revenue = res_orders['quantity'] * res_orders['item_price']

In [44]:
revenue

0        2.39
1        3.39
2        3.39
3        2.39
4       33.96
        ...  
4617    11.75
4618    11.75
4619    11.25
4620     8.75
4621     8.75
Length: 4622, dtype: float64

In [45]:
total_revenue = revenue.sum()
total_revenue

39237.02

* Total revenue: \$39237

##### Q) How many orders were made in the period?

In [46]:
res_orders

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  
0                                                   NaN        2.39  
1                                          [Clementine]        3.39  
2                                               [Apple]        3.39  
3                                                   NaN        2.39  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  
...                                                 ...         ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...       11.75  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       11.75  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...       11.25  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...        8.75  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...        8.75  

[4622 rows x 5 columns]

In [47]:
total_orders = res_orders['order_id'].nunique()


In [48]:
res_orders['order_id'].value_counts().count()

1834

* 1834 total orders were placed

##### Q) What is the average revenue amount per order?

In [49]:
total_revenue/total_orders

21.39423118865867

* Average revenue amount per order is \$21.39

##### Q) How many different items are sold?

In [50]:
res_orders

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  
0                                                   NaN        2.39  
1                                          [Clementine]        3.39  
2                                               [Apple]        3.39  
3                                                   NaN        2.39  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  
...                                                 ...         ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...       11.75  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       11.75  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...       11.25  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...        8.75  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...        8.75  

[4622 rows x 5 columns]

In [51]:
res_orders['item_name'].nunique()

50

In [52]:
res_orders['item_name'].value_counts()

Chicken Bowl                             726
Chicken Burrito                          553
Chips and Guacamole                      479
Steak Burrito                            368
Canned Soft Drink                        301
Chips                                    211
Steak Bowl                               211
Bottled Water                            162
Chicken Soft Tacos                       115
Chicken Salad Bowl                       110
Chips and Fresh Tomato Salsa             110
Canned Soda                              104
Side of Chips                            101
Veggie Burrito                            95
Barbacoa Burrito                          91
Veggie Bowl                               85
Carnitas Bowl                             68
Barbacoa Bowl                             66
Carnitas Burrito                          59
Steak Soft Tacos                          55
6 Pack Soft Drink                         54
Chips and Tomatillo Red Chili Salsa       48
Chicken Cr

In [53]:
res_orders['item_name'].value_counts().count()

50

In [54]:
len(res_orders['item_name'].value_counts())

50

* 50 different items are sold